smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/IncomingRabbitMQMetadata.java

```diff
@@ -20,19 +22,25 @@
  */
 public class IncomingRabbitMQMetadata {
 
-    private final RabbitMQMessage message;
     private final Map<String, Object> headers;
+    private final BasicProperties properties;
+    private final Envelope envelope;
 
     /**
      * Constructor.
      *
      * @param message the underlying {@link RabbitMQMessage}
      */
     IncomingRabbitMQMetadata(RabbitMQMessage message) {
-        this.message = message;
+        this(message.properties(), message.envelope());
+    }
+
+    IncomingRabbitMQMetadata(BasicProperties properties, Envelope envelope) {
+        this.properties = properties;
```

Comment:
Are we sure that this cannot be `null`? I see several accesses in the other methods.

Note that it's not new, it was already like this before.

Reply:
Did not think about that, do you have a standard for how you handle this? I can add `Objects.requireNonNull()` in constructor to ensure NPE on creation.